File: Chapter09/lazyEvaluation.clj
; Written by Raju
; Run using following command
; clojure lazyEvaluation.clj
; Lazy Evaluation in Clojure
(def lazyVal (delay 10))
(println lazyVal)
(println (realized? lazyVal))
(println (class lazyVal))
(println (deref lazyVal))
(println lazyVal)
(println (realized? lazyVal))
(println (def lazyVal (delay 10)))

(println (force lazyVal))
(println (def lazyVal (delay 10)))
(println @lazyVal)
(println (def fp (promise)))
(println (realized? fp))
(println (deliver fp (str "MyString")))
(println (realized? fp))
(println fp)
(println (deref fp))
(println (def lazyDate ( delay (new java.util.Date))))
(println lazyDate )
(println (def simpleDate  (new java.util.Date)))
(println simpleDate)
(println (count [1,2,3,4,5]))
(println simpleDate)
(println (deref lazyDate))

File: Chapter09/lazyEvaluation.scala
// Written by Raju
// Run it using following command
// scala lazyEvaluation.scala
var sv = 5
sv = 6
val svc = 5
sv = 5
lazy val svl = 5
// Lazy Value date example 
import  java.util.Calendar  //Line One
lazy val lazyDate = Calendar.getInstance.getTime  //Line Two
val simpleDate = Calendar.getInstance.getTime  // Line Three
val rl = 1 to 20 //Line Four 
for(x <- rl) print(x) //Line Five 
 println("Simple Date Variable Value Is : " + simpleDate) // Line Six
 println("Lazy Date Variable Value Is : " + lazyDate) // Line Seven

